src/sagas/pushNotification.js

```diff
@@ -0,0 +1,412 @@
+      yield put(pushRegisterFailed());
+    }
+  } catch (error) {
+    console.log('Error registering device for the first time: ', error.cause);
+    yield put(pushRegisterFailed());
+  }
+}
+
+/**
+ * This function is responsible for registering the device on the wallet-service in the event
+ * of renewing the registration.
+ */
+export function* registration({ payload: { enabled, showAmountEnabled, deviceId } }) {
+  yield put(pushLoadWalletRequested());
+
+  // wait for the wallet to be loaded
+  const [loadWalletSuccess, loadWalletFail] = yield race([
+    take(types.PUSH_WALLET_LOAD_SUCCESS),
+    take(types.PUSH_WALLET_LOAD_FAILED)
+  ]);
+
+  if (loadWalletFail) {
+    yield put(pushRegisterFailed());
+  }
+
+  const { walletService } = loadWalletSuccess.payload;
+  try {
+    const { success } = yield call(pushLib.PushNotification.registerDevice, walletService, {
+      pushProvider: pushLib.PushNotificationProvider.ANDROID,
+      deviceId,
+      enablePush: true,
+    });
+
+    if (success) {
+      const enabledAt = Date.now();
+      STORE.setItem(pushNotificationKey.enabledAt, enabledAt);
+      yield put(pushRegisterSuccess({ enablePush: enabled, enableShowAmounts: showAmountEnabled, hasBeenEnabled: true, enabledAt }));
+    } else {
+      // NOTE: theoretically, this should never happen because when the client call fails, it throws an error
+      yield put(pushRegisterFailed());
+    }
+  } catch (error) {
+    console.log('Error registering device: ', error.cause);
+    yield put(pushRegisterFailed());
+  }
+}
+
+/**
+ * This function is responsible for updating the registration of the device on the wallet-service
+ * in the event of changing the settings of the push notifications.
+ * @param {{ payload: { enabled: boolean, showAmountEnabled: boolean, deviceId: string } }} action
+ */
+export function* updateRegistration({ payload: { enabled, showAmountEnabled, deviceId } }) {
+  yield put(pushLoadWalletRequested());
+
+  // wait for the wallet to be loaded
+  const [loadWalletSuccess, loadWalletFail] = yield race([
+    take(types.PUSH_WALLET_LOAD_SUCCESS),
+    take(types.PUSH_WALLET_LOAD_FAILED)
+  ]);
+
+  if (loadWalletFail) {
+    yield put(pushRegisterFailed());
+  }
+
+  const { walletService } = loadWalletSuccess.payload;
+  try {
+    const { success } = yield call(pushLib.PushNotification.updateDevice, walletService, {
+      deviceId,
+      enablePush: enabled,
+      enableShowAmounts: showAmountEnabled,
+    });
+    if (success) {
+      yield put(pushUpdateSuccess({ enabled, showAmountEnabled }));
+    } else {
+      // NOTE: theoretically, this should never happen because when the client call fails, it throws an error
+      yield put(pushUpdateFailed());
+    }
+  } catch (error) {
+    console.log('Error updating device: ', error.cause);
+    yield put(pushUpdateFailed());
+  }
+}
+
+export function* saga() {
+  yield all([
+    takeEvery(types.PUSH_INIT, appInitialization),
```

Comment:
The challenge of working with Effects is that we can lose some processes from sight. A future maintainer will have no clue that the push notification was initialized when the wallet initializes as well because it will be hidden in the saga middleware.

The way I did the process is explicit in the app initialization, signaling that the push notification exists and the app initialization is the entry point.